src/with-hover.js

```diff
@@ -21,6 +22,12 @@ const withHover = ({
     onMouseLeave: PropTypes.func,
   }
 
+  constructor(props) {
+    super(props);
+    this.onMouseEnter = this.onMouseEnter.bind(this)
```

Comment:
Just a quick question, could this be fixed instead of writing like it is in the code:
```
 onMouseEnter = () => {
    if (!this.state.hover) { this.setState({ hover: true }) }
  }
```
Writing it as a method like:

```
 onMouseEnter() {
    if (!this.state.hover) { this.setState({ hover: true }) }
  }
```

Reply:
Just a thought. I imagine but can't be certain of the intent of the fix.

Reply:
Hi!

The issue is that the context of onMouseEnter and onMouseEnter becomes lost in the click handler which means that this becomes null in this case.

This was not occurring in previous versions of react as it was done automatically. This is unfortunately no longer the case as of version 16.

We can either use a clojure, use a lambda in the props (bad for performance) or bind the context of the method in the class constructor to solve this issue. Unfortunately using the lamda syntax and the method syntax is equivalent in this context and it still needs binding.